python/lsst/daf/butler/datastores/webdavDatastore.py

```diff
@@ -0,0 +1,271 @@
+
+        serializedDataset = response.content
+
+        formatter = getInfo.formatter
+        try:
+            result = formatter.fromBytes(serializedDataset,
+                                         component=getInfo.component if isComponent else None)
+        except NotImplementedError:
+            log.debug("Formatter does not have extension, using temporary file.")
+            tmpLoc = Location(".", "temp")
+            tmpLoc = formatter.makeUpdatedLocation(tmpLoc)
+            with tempfile.NamedTemporaryFile(suffix=tmpLoc.getExtension()) as tmpFile:
+                tmpFile.write(serializedDataset)
+                tmpFile.flush()
+                formatter._fileDescriptor.location = Location(*os.path.split(tmpFile.name))
+                result = formatter.read(component=getInfo.component if isComponent else None)
+        except Exception as e:
+            raise ValueError(f"Failure from formatter '{formatter.name()}' for dataset {ref.id}"
+                             f" ({ref.datasetType.name} from {location.uri}): {e}") from e
+
+        return self._post_process_get(result, getInfo.readStorageClass, getInfo.assemblerParams,
+                                      isComponent=isComponent)
+
+    def _write_in_memory_to_artifact(self, inMemoryDataset: Any, ref: DatasetRef) -> StoredFileInfo:
+        location, formatter = self._prepare_for_put(inMemoryDataset, ref)
+
+        if webdavCheckFileExists(location, session=self.session,)[0]:
+            raise FileExistsError(f"Cannot write file for ref {ref} as "
+                                  f"output file {location.uri} exists.")
+
+        try:
+            serializedDataset = formatter.toBytes(inMemoryDataset)
+            self.session.put(location.uri.geturl(), data=serializedDataset)
+            log.debug("Wrote file directly to %s", location.uri)
+        except NotImplementedError:
+            with tempfile.NamedTemporaryFile(suffix=location.getExtension()) as tmpFile:
+                formatter._fileDescriptor.location = Location(*os.path.split(tmpFile.name))
+                formatter.write(inMemoryDataset)
+                with open(tmpFile.name, 'rb') as f:
+                    self.session.put(location.uri.geturl(), data=f)
+                log.debug("Wrote file to %s via a temporary directory.", location.uri)
+
+        if self._transaction is None:
+            raise RuntimeError("Attempting to write artifact without transaction enabled")
+
+        self._transaction.registerUndo("write", webdavDeleteFile, location)
+
+        return self._extractIngestInfo(location.uri.geturl(), ref, formatter=formatter)
+
+    def _overrideTransferMode(self, *datasets: Any, transfer: Optional[str] = None) -> Optional[str]:
+        if transfer != "auto":
+            return transfer
+        return "copy"
+
+    def _standardizeIngestPath(self, path: str, *, transfer: Optional[str] = None) -> str:
```

Comment:
This should now be identical to the s3 version. We can probably consider moving the s3 one to fileLikeDatastore.py and not having a webdav one.